store/searchlayer/user_layer.go

```diff
@@ -37,7 +37,7 @@ func (s *SearchUserStore) deleteUserIndex(user *model.User) {
 func (s *SearchUserStore) Search(teamId, term string, options *model.UserSearchOptions) ([]*model.User, error) {
 	for _, engine := range s.rootStore.searchEngine.GetActiveEngines() {
 		if engine.IsSearchEnabled() {
-			listOfAllowedChannels, nErr := s.getListOfAllowedChannelsForTeam(teamId, options.ViewRestrictions)
+			listOfAllowedChannels, nErr := s.getListOfAllowedChannelsForTeam(teamId, "", options.ViewRestrictions)
```

Comment:
By definition DMs and GMs have no team so this doesn't make too much sense in my mind. I understand it may be the quickest way to achieve what we need here but I'd consider a refactoring and likely provide either something more abstract or a dedicated method for such channels.

Reply:
Makes sense. I can rename it to `getListOfAllowedChannels`. I think in any case it will become something like that.
Either we do `getListOfAllowedChannels` -> ForTeam() + AddDMGMChannel(), or we rename this and keep it like this.

Any preference?

Reply:
If there are no immediate performance concerns I'd probably go with separate methods since it seems like this would be doing too many things that are almost unrelated given the arguments we are passing.

Reply:
Sounds good

Reply:
@streamer45 - Took a deeper look. Separate methods also has some issues. We want to add the channel only when `if teamId != "" && (viewRestrictions == nil || strings.Contains(strings.Join(viewRestrictions.Teams, "."), teamId)) {` evaluates to true. So that means I need to copy over the logic to a separate method. Alternatively, I can create a dedicated store method which returns all team channels + the passed channel if it's a DM/GM channel.

Both alternatives seem a bit complicated to me compared to the option of just renaming the original method.

Let me know what you think.

Reply:
I am not sure I understand why we need to check the team to fetch DMs/GMs. Shouldn't they be decoupled from that?

Reply:
Discussed internally. This piece of code would need some serious refactoring that goes outside the scope of fixing the issue at hand. I'd still suggest we rename the method to make it a little less inaccurate but not a blocker at this point.